server/manager/src/main/java/org/apache/accumulo/manager/ManagerClientServiceHandler.java

```diff
@@ -89,13 +93,22 @@
 import org.apache.zookeeper.KeeperException.NoNodeException;
 import org.slf4j.Logger;
 
+import com.github.benmanes.caffeine.cache.Cache;
+import com.github.benmanes.caffeine.cache.Caffeine;
+import com.github.benmanes.caffeine.cache.Weigher;
+
 public class ManagerClientServiceHandler implements ManagerClientService.Iface {
 
   private static final Logger log = Manager.log;
   private final Manager manager;
 
+  private final Cache<KeyExtent,Long> recentHostingRequest;
+
   protected ManagerClientServiceHandler(Manager manager) {
     this.manager = manager;
+    Weigher<KeyExtent,Long> weigher = (extent, t) -> Splitter.weigh(extent) + 8;
+    this.recentHostingRequest = Caffeine.newBuilder().expireAfterWrite(1, TimeUnit.MINUTES)
+        .maximumWeight(10_000_000L).weigher(weigher).build();
```

Comment:
Is this a max of 10M entries?

Reply:
Should be a max mem of 10M not including overhead of java objects.

Reply:
Ok, I wasn't sure what the Weigher and maxWeight meant. I think it would be more clear if we did something like:
```
  private static final long TEN_MB = 10 * 1024 * 1024;
```

and then use TEN_MB as the method parameter.